File: RealEstateApi/Models/Review.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;

public class Review
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    public Guid TargetUserId { get; set; } 

    [Required]
    public Guid ReviewerId { get; set; }
    [Required]

    [Range(1, 5, ErrorMessage = "Điểm đánh giá phải từ 1 đến 5")]
    public int Rating { get; set; }

    [StringLength(500, ErrorMessage = "Nhận xét không vượt quá 500 ký tự")]
    public string? Comment { get; set; }
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [JsonIgnore]
    public virtual User? Reviewer { get; set; }

    [JsonIgnore]
    public virtual User? TargetUser { get; set; }

}

File: RealEstateFrontend/src/models/Review.js
import { v4 as uuidv4 } from 'uuid';

export class Review {
    constructor(id = uuidv4(), propertyId, userId, rating, comment, createdAt) {
        this.id = id;
        this.propertyId = propertyId;
        this.userId = userId;
        this.rating = rating;
        this.comment = comment;
        this.createdAt = createdAt;
    }
}
